test/integration/targets/postgresql/tasks/unsorted.yml

```
@@ -0,0 +1,789 @@
+#
+# Create and destroy db
+#
+- name: Create DB
+  become_user: "{{ pg_user }}"
+  become: yes
+  postgresql_db:
+    state: present
+    name: "{{ db_name }}"
+    login_user: "{{ pg_user }}"
+  register: result
+
+- name: assert that module reports the db was created
+  assert:
+    that:
+       - result is changed
+       - "result.db =='{{ db_name }}'"
```

The `{{ }}` shouldn't be necessary here. This seems to be comparing to the literal string `{{ db_name }}`.
```suggestion
       - "result.db == db_name"
```
If you need the `db_name` in single quotes.
```suggestion
       - 'result.db == "''" ~ db_name ~ "''"'
```

Or just strip the quotes from `result.db`:
```suggestion
       - "result['db'].strip("''") == db_name'"
```

---

There is a lot of possibilities to write the same things in different ways, so I'm not sure it is good or bad :) sometimes the syntax may be confused.
Thanks for the examples, added to the check list as always :)
done